client/screens/ProfilePage.js

```diff
@@ -42,23 +51,68 @@ const ProfilePage = (props) => {
   const userRef = ref(db, "users/" + props.userId);
   const newUserRef = push(userRef);
 
+  const storage = getStorage();
+
   // Image Picker
   const pickImage = async () => {
     // No permissions request is necessary for launching the image library
-    let result = await ImagePicker.launchImageLibraryAsync({
+    let pickerResult = await ImagePicker.launchImageLibraryAsync({
       mediaTypes: ImagePicker.MediaTypeOptions.All,
       allowsEditing: true,
       aspect: [4, 3],
       quality: 1,
     });
+    // console.log(pickerResult);
+
+    handleImagePicked(pickerResult);
 
-    console.log(result);
+    if (!pickerResult.cancelled) {
+      setImage(pickerResult.uri);
+    }
+  };
 
-    if (!result.cancelled) {
-      setImage(result.uri);
+  const handleImagePicked = async (pickerResult) => {
+    try {
+      setUploading(true);
+
+      if (!pickerResult.cancelled) {
+        const uploadUrl = await uploadImageAsync(pickerResult.uri);
+        setImage(uploadUrl);
+      }
+    } catch (e) {
+      // console.log(e);
+      alert("Upload failed, sorry :(");
+    } finally {
+      setUploading(false);
     }
   };
 
+  async function uploadImageAsync(uri) {
+    // Why are we using XMLHttpRequest? See:
+    // https://github.com/expo/expo/issues/2402#issuecomment-443726662
+    const blob = await new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.onload = function () {
+        resolve(xhr.response);
+      };
+      xhr.onerror = function (e) {
+        // console.log(e);
+        reject(new TypeError("Network request failed"));
+      };
+      xhr.responseType = "blob";
+      xhr.open("GET", uri, true);
+      xhr.send(null);
+    });
+
+    const fileRef = sRef(getStorage(), uuid.v4());
+    const result = await uploadBytes(fileRef, blob);
+
+    // We're done with the blob, close and release it
+    // blob.close();
```

Comment:
Do we still need this code here?

Reply:
Yes just commented it out since it was breaking the app but will probably be used for the next part of this functionality once I figure out the issue.